### PR TITLE
Add sampling masks (MagicMask, VariableDensityPoissonMask)

### PR DESCRIPTION
**Pull request by georgeyiasemis**
Adding:
* Variable density Poisson masking function
  * cython implementation
  * setup.py changes 
* Fastmri's Magic masking function
* Additional tests
Update:
* `torch.where` output needed to be made `contiguous` to be inputted to fft, due to new torch version

### CODE REVIEW

**Review by georgeyiasemis — COMMENTED** [review]
OK

**Comment by georgeyiasemis** [comment]
@jonasteuwen The issue was fixed, so you can now review.

**Review by georgeyiasemis — COMMENTED** [review]
OK

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -21,5 +21,6 @@ jobs:
       run: |
         python -m pip install --upgrade pip setuptools wheel
         pip install tox tox-gh-actions
+        python3 -m pip install -e ".[dev]"
```
Comment (jonasteuwen) [inline .github/workflows/tox.yml:24]: This should not be required. That's what tox should be doing

File: direct/common/_poisson.pyx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+cdef inline Py_ssize_t fmin(Py_ssize_t one, Py_ssize_t two) nogil:
+    """Min(a, b)."""
+    return one if one < two else two
+
+
+def poisson(
+    int nx,
+    int ny,
+    int max_attempts,
+    cnp.ndarray[cnp.int_t, ndim=2, mode='c'] mask,
```
Comment (jonasteuwen) [inline direct/common/_poisson.pyx:41]: This is based on some code right? You should reference this, and see if the licenses are compatible

**Review by jonasteuwen — COMMENTED** [review]
Small comments

File: .github/workflows/tox.yml (hunk 1 above)
Comment (georgeyiasemis) [reply .github/workflows/tox.yml]: Fixed

**Review by jonasteuwen — APPROVED** [review]
LGTM
